apps/openmw/mwinput/inputmanagerimp.cpp

```diff
@@ -1014,10 +1016,33 @@ namespace MWInput
 
     void InputManager::screenshot()
     {
-        mScreenCaptureHandler->setFramesToCapture(1);
-        mScreenCaptureHandler->captureNextFrame(*mViewer);
+        bool regularScreenshot = true;
 
-        MWBase::Environment::get().getWindowManager()->messageBox ("Screenshot saved");
+        std::string settingStr;
+
+        try
+        {
+            settingStr = Settings::Manager::getString("screenshot type","Video");
+            regularScreenshot = settingStr.size() == 0;
+        }
+        catch (std::runtime_error)
+        {
+        }
+
+        if (regularScreenshot)
+        {
+            mScreenCaptureHandler->setFramesToCapture(1);
+            mScreenCaptureHandler->captureNextFrame(*mViewer);
+            MWBase::Environment::get().getWindowManager()->messageBox("Screenshot saved");
+        }
+        else
+        {
+            osg::ref_ptr<osg::Image> screenshot (new osg::Image);
+
+            if (MWBase::Environment::get().getWorld()->screenshot360(screenshot.get(),settingStr))
+                (*mScreenCaptureOperation) (*(screenshot.get()),0);
+                // mScreenCaptureHandler->getCaptureOperation() causes crash for some reason
```

Comment:
Any idea why `mScreenCaptureHandler->getCaptureOperation()` causes segfault? If I was able to do it I'd be able to modify fewer files. Now I have to awkwardly pass the operation itself.

Reply:
It's not evel that `getCaptureOperation()` returns 0, it's the call itself that causes the crash (`mScreenCaptureHandler` isn't 0 either). Is it OSG bug then?

Reply:
So this should be my last issue ATM.

Reply:
There is a [const and non-const](https://github.com/openscenegraph/OpenSceneGraph/blob/3b1212fed59b1024aab289bf7837056c559913e1/src/osgViewer/ScreenCaptureHandler.cpp) version of `getCaptureOperation`, neither works.

Reply:
Just look at the (rather strange) [implementation](https://github.com/openscenegraph/OpenSceneGraph/blob/master/src/osgViewer/ScreenCaptureHandler.cpp#L54) and you'll see. Using ->_defaultCaptureOperation instead should work.

Reply:
No, we can't use that because it's in a private callback. Too bad. I don't see any solution.